app/js/components/names.jsx

```diff
@@ -1,265 +1,251 @@
- * compliance with the License. You may obtain a copy of the License at
- * http://license.openmrs.org
- * Software distributed under the License is distributed on an "AS IS"
- * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
- * License for the specific language governing rights and limitations
- * under the License.
- * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+/* * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
  */
 import React from 'react';
 import apiCall from '../utilities/apiHelper';
 
 class Name extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            display: 'view',
-            givenName: '',
-            middleName: '',
-            familyName: '',
-            voided: false,
-            preferred: true,
-            creator: '',
-            dateCreated: '',
-            res: '',
-        };
-        this.handleEdit = this.handleEdit.bind(this);
-        this.handleSave = this.handleSave.bind(this);
-        this.handleCancel = this.handleCancel.bind(this);
-        this.handleChange = this.handleChange.bind(this);
-        this.fetch = this.fetch.bind(this);
-    }
-
-    /**
-     *
-     * set to state patient attributes
-     *
-     * @memberOf Names
-     */
-    componentWillMount() {
-        this.fetch();
+  constructor(props) {
+    super(props);
+    this.state = {
+      display: 'view',
+      givenName: '',
+      middleName: '',
+      familyName: '',
+      voided: false,
+      preferred: true,
+      creator: '',
+      dateCreated: '',
+      res: '',
+      error: ''
     };
-
-    /**
-     *
-     * make API call to get the values then set to state
-     *
-     * @memberOf Names
-     */
-    fetch(){
-        apiCall(null, 'get', 'patient/' + this.props.uuid + '?v=full').then((res) => {
-          this.setState({   res: res.person,
-              givenName: res.person.preferredName.givenName,
-              middleName: res.person.preferredName.middleName,
-              familyName: res.person.preferredName.familyName,
-              voided: res.person.preferredName.voided,
-              creator: res.person.auditInfo.creator.display,
-              dateCreated: res.person.auditInfo.dateCreated
-            });
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.fetch = this.fetch.bind(this);
+  }
+
+  componentWillMount() {
```

Comment:
Is componentWillMount not expensive? Can we have it at componentDidMount

Reply:
Thanks for this, had not yet understood the implications of the different lifecycle stages